src/prefect/client/orion.py

```diff
@@ -962,6 +962,16 @@ async def create_block_document(
                 `SecretBytes` fields. Note Blocks may not work as expected if
                 this is set to `False`.
         """
+        if isinstance(block_document, BlockDocument):
+            block_document_kwargs = block_document.dict(
+                json_compatible=True,
+                include_secrets=include_secrets,
+                exclude_unset=True,
+                exclude={"id", "block_schema", "block_type"},
+            )
+            block_document = schemas.actions.BlockDocumentCreate(
+                **block_document_kwargs
+            )
```

Comment:
Can you use `schemas.actions.BlockDocumentCreate.parse_obj(...)`? (actually not sure)

Should probably update the function signature to indicate this is supported as well.